lib/mongo_ecto/encoder.ex

```diff
@@ -46,6 +46,8 @@ defmodule Mongo.Ecto.Encoder do
       :error       -> :error
     end
   end
+  def encode(%Mongo.Ecto.DiffEmbed{} = diff, _pk),
```

Comment:
To be clear: my point is that you already have a clause for handling DiffEmbed in the encoder, it would be nice to not have another clause in the normalized query code.